### PR TITLE
Issue #9803: EmptyLineSeparator fix message and documentation

### LINKED ISSUE #9803: EmptyLineSeparator: fix message and documentation
detected at https://github.com/checkstyle/checkstyle/pull/9707#issuecomment-812521756

TODO:
1) update message 
`empty.line.separator=''{0}'' should be separated from previous statement.`
to mention "comments" or "from previous line" or create new message. It should be some thing like `''{0}'' should be separated by empty line from next block or statement or comment``

2) update default example with code I provided in PR comment

3) description of Check should explain that we do validation not for "after" but from current line to previous line and there might be comments/blocks/.../anything (not only statements)

### PR DESCRIPTION
**Pull request by aryaniiit002**
resolves #9803

### CODE REVIEW

**Comment by aryaniiit002** [comment]
GitHub, generate web site

**Review by romani — APPROVED** [review]

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/whitespace/EmptyLineSeparatorCheck.java [hunk 1]
```diff
@@ -209,6 +213,26 @@
...
  * &lt;/module&gt;
  * </pre>
+ * <pre>
+ * ///////////////////////////////////////////////////
+ * //HEADER
+ * ///////////////////////////////////////////////////
+ *
+ *
+ * package com.checkstyle.whitespace; // violation, 'package' more than 1 empty lines before.
+ *
```
Comment (aryaniiit002) [inline src/main/java/com/puppycrawl/tools/checkstyle/checks/whitespace/EmptyLineSeparatorCheck.java:223]: Seen, i will correct this 
'package' `has` more

Comment (aryaniiit002) [reply src/main/java/com/puppycrawl/tools/checkstyle/checks/whitespace/EmptyLineSeparatorCheck.java]: rebased to updated master

Comment (aryaniiit002) [reply src/main/java/com/puppycrawl/tools/checkstyle/checks/whitespace/EmptyLineSeparatorCheck.java]: rebased to updated master

**Review by pbludov — APPROVED** [review]

**Comment by pbludov** [comment]
@aryaniiit002 please rebase PR to the latest master

**Comment by aryaniiit002** [comment]
@pbludov it's done

**Comment by pbludov** [comment]
> @pbludov it's done

The last commit is fac0e580ae271fb72fff8126de6bfc73425ef8f4
https://github.com/checkstyle/checkstyle/pull/9845/commits

Maybe you forgot to `git push --force`?

**Comment by aryaniiit002** [comment]
```
aryaniiit002@ENIAC:~/Desktop/checkstyle$ git push --force origin EmptyLineSeparator_Branch
Username for 'https://github.com': aryaniiit002
Password for 'https://aryaniiit002@github.com': 
Everything up-to-date

```

**Comment by aryaniiit002** [comment]
i think we have some collateral damage = Pitest1

**Comment by aryaniiit002** [comment]
https://github.com/checkstyle/checkstyle/blob/0ca66ad6a7e1bdc74fb1189363c69dfee60794b9/.ci/pitest.sh#L106
can i remove this line and rebase the commit into one?

**Comment by aryaniiit002** [comment]
@pbludov rebased again, please proceed

**Pull request merged**